sql/expression/in.go

```diff
@@ -229,20 +230,17 @@ func newInMap(expr sql.Expression, lType sql.Type) (map[uint64]sql.Expression, b
 	switch right := expr.(type) {
 	case Tuple:
 		for _, el := range right {
-			switch l := el.(type) {
-			case *Literal, Tuple:
-				key, err := hashOf(l, lType)
-				if sql.ErrInvalidType.Is(err) {
-					// TODO: can't convert a tuple in right expr to left literal type, and vice versa, echo warning?
-					continue
-				}
-				if err != nil {
-					return nil, hasNull, err
-				}
-				elements[key] = el
-			default:
-				return nil, hasNull, ErrUnsupportedHashInSubexpression.New(el)
+			if !IsFolded(el) {
+				return nil, hasNull, ErrHashInRightNotFolded.New(el)
```

Comment:
Should be impossible right? Only if there's an error in the analyzer?

Would add a note to that effect